src/geometry.h

```diff
@@ -136,19 +136,49 @@ namespace Faunus {
                 void to_json(json &j) const;
 
                 inline void boundary( Point &a ) const override {
-                    if ( std::fabs(a.x()) > len_half.x())
+		    if ( type == CUBOID or type == SPHERE or type == CYLINDER or type == SLIT ) {
+		      if ( std::fabs(a.x()) > len_half.x())
                         a.x() -= len.x() * anint(a.x() * len_inv.x());
 
-                    if ( std::fabs(a.y()) > len_half.y())
+		      if ( std::fabs(a.y()) > len_half.y())
                         a.y() -= len.y() * anint(a.y() * len_inv.y());
 
-                    if ( std::fabs(a.z()) > len_half.z())
+		      if ( std::fabs(a.z()) > len_half.z())
                         a.z() -= len.z() * anint(a.z() * len_inv.z());
+		    } else if( type == HEXAGONAL ) {
+		      Point unitvX = Point(1.0,0.0,0.0);
+		      Point unitvY = Point(0.5,sqrt(3.0)/2.0,0.0);
+		      Point unitvZ = Point(-0.5,sqrt(3.0)/2.0,0.0);
+		      if(a.dot(unitvX) > len_half.x())
+			a = a - len.x()*unitvX;
+		      if(a.dot(unitvX) < -len_half.x())
```

Comment:
evaluate `a.dot(unitvX` etc only once and save

Reply:
Can not do this since 'a' is changing after each evaluated dot-product.